Add test for TreeLineTracker.getLineInformation

lacinoire opened the pull request:
Hey 😊
I want to contribute the following test:

Test that a `org.eclipse.lemminx.commons.BadLocationException` is thrown when the parameter `line` of `Position.<init>` is set to `-1`.
This tests the method [`TreeLineTracker.getLineInformation`](https://github.com/eclipse/lemminx/blob/9e879e3de0c363e12651f3aa0cdc82a050d79143/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/TreeLineTracker.java#L1281).
This test is based on the test [`testEmptyDocumentInc`](https://github.com/eclipse/lemminx/blob/9e879e3de0c363e12651f3aa0cdc82a050d79143/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/commons/TextDocumentTest.java#L144).

Curious to hear what you think!

Also, is the description I provided of the test is helping you to answer to this pull request? Why (not)?

(I wrote this test as part of a research study at TU Delft. [Find out more](https://github.com/lacinoire/lacinoire/blob/main/README.md))

rgrunber commented on org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/commons/TextDocumentTest.java:
```diff
@@ -155,6 +156,18 @@ public void testEmptyDocumentInc() throws BadLocationException {
+				() -> {
+					TextDocument document = new TextDocument("", "");
+					document.setIncremental(true);
+					Position position = document.positionAt(0);
```
Also just hit me that `positionAt` also throws `BadLocationException`, and if that did happen, this would just pass. I would just remove this line and have `Position position = new Position(-1, 0);`

datho7561 commented on org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/commons/TextDocumentTest.java:
```diff
@@ -155,6 +156,18 @@ public void testEmptyDocumentInc() throws BadLocationException {
 
+	@Test
+	public void testGetLineInformation() throws BadLocationException {
+		Assertions.assertThrows(BadLocationException.class,
```
Just another small thing: we tend to do static imports for the JUnit assert methods (see the imports for `assertNotEquals` and `assertNotNull`).

datho7561 reviewed (changes requested):
Thanks for the contribution! Once you've addressed Roland's comments, I think it should be good to merge.

lacinoire commented:
Thank you for the detailed feedback! 😄
I've incorporated all three changes as suggested.


For the formatting @rgrunber : 
I'm using IntelliJ. Though the `() -> {` was already put on a new line during the test creation (by the pretty-printer of [Spoon](https://github.com/INRIA/spoon) afaik), so I assume that confused IntelliJ into the large indent.
Changed it manually to your suggestions & fits better with the rest of the class now 🙂

angelozerr commented on org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/commons/TextDocumentTest.java:
```diff
@@ -155,6 +156,16 @@ public void testEmptyDocumentInc() throws BadLocationException {
 	}
 
+	@Test
+	public void testGetLineInformation() throws BadLocationException {
```
I will rename this test with `badLocation`

angelozerr commented:
Thanks so much @lacinoire for your contribution!
